config/extension_hook.go

```diff
@@ -69,7 +82,19 @@ func (hook *Hook) run(args ...interface{}) (result []interface{}, err error) {
 		return
 	}
 
+	// Add global environment variables to the current context
+	// (these variables will be available while and after the execution of the hook)
+	for key, value := range hook.GlobalVars {
```

Comment:
Both of these behaviors really need a functional test. As it is, I wouldn't be confident to modify any of this without fear of breaking it